service/catalog.go

```diff
@@ -84,16 +139,28 @@ func updateCatalog(w http.ResponseWriter, r *http.Request, envId string) (int, e
 		return http.StatusBadRequest, err
 	}
 
-	if err := db.Model(&model.CatalogModel{}).Where(&model.CatalogModel{
+	if isDuplicateGlobalName(catalogModel) {
+		return http.StatusUnprocessableEntity, errors.New("Cannot add catalog with duplicate field `name`")
```

Comment:
So, you're checking if the new catalog name exists as global. What if the new catalog name already exists and it is a different catalog....this would wipe out that catalog. Is that the behavior we want?